aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -639,22 +696,15 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
         // Replace attn_mask with causal mask; lower triangular elements take part in attention.
         const auto L = query.sym_size(-2), S = key.sym_size(-2);
         attn_mask = at::ones_symint({L, S}, query.options().dtype(at::kBool)).tril();
+        attn_mask = convert_boolean_attn_mask(attn_mask, query.dtype());
```

Comment:
nit: Similar story here where the mask was just allocated and now getting reallocated again. [`triu`](https://pytorch.org/docs/master/generated/torch.triu.html#torch.triu) should also work for float so if `convert_boolean_attn_mask` returns a fully `-inf`'d attn_mask you could use triu to cut off the top part.